app/controllers/holds_controller.rb

```diff
@@ -35,6 +37,56 @@ def new
     @bib = Bib.new(parsed_body, holdable_locations)
   end
 
+  # Handles placing holds
+  #
+  # POST /holds
+  def create
+    results = barcodes.each_with_object(success: [], error: []) do |barcode, status|
+      hold_args = { pickup_library: params['pickup_library'], pickup_by_date: params['pickup_by_date'] }
+      response = symphony_client.place_hold(patron,
+                                            current_user.session_token,
+                                            barcode,
+                                            hold_args)
+      case response.status
+      when 200
+        status[:success] << { barcode: barcode,
+                              hold_key: JSON.parse(response.body).dig('holdRecord', 'key') }
+      else
+        Rails.logger.error("Place Hold for #{barcode} by #{patron.barcode}: #{response.body}")
+        status[:error] << { barcode: barcode,
+                            error_message: JSON.parse(response.body).dig('messageList')[0].dig('message') }
+      end
+    end
+
+    session[:place_hold_results] = results
```

Comment:
Cookie overflow still is an issue here, right? We're going to look at this across the board with #77 right? Starting to think adding a redis might be worth looking at. Take a look at [this blogpost](https://medium.com/@kirill_shevch/configuration-cache-and-rails-session-store-with-redis-b3ce6f64d1fc)?

Reply:
yeah, would be an issue esp if we were to save the whole objects to the session, here I tried to go with the minimum info possible like hold key, barcode and error message if there is one. 

I will take a look at that blogpost, thanks for sharing.

Reply:
RIght, so it isn't likely to cause an overflow. But, in general, if we went with an alternative session store like Redis it would give us piece of mind across the application. Doesn't seem to difficult to implement either, just kinda drop-in with a gem and configure it.

Reply:
Not likely but still we can not guarantee that it won't, I can think of a case for somebody trying to put a hold to many volumes of a record which might trigger the overflow issue. So yes, I agree that we need to look into Redis, also considering that this issue might happen for different actions across the application.